packages/vite/src/node/ssr/ssrModuleLoader.ts

```diff
@@ -257,7 +257,11 @@ async function nodeImport(
   try {
     const mod = await dynamicImport(url)
     return proxyESM(mod)
-  } catch {}
+  } catch (err) {
+    err.message = `\nFailed to import ${url} from ${importer}\n${err.message}`
+
+    throw err
+  }
```

Comment:
Do we want this catch block? (I prefer the error message without the catch block.)

**With**
```
❯ node test.js
Error: 
Failed to import file:///home/vitejs/stackblitz-reproductions/bluwy-create-vite-extra-56u3yb/node_modules/.pnpm/astro-integration-lottie@0.2.0/node_modules/astro-integration-lottie/dist/index.js from /home/vitejs/stackblitz-reproductions/bluwy-create-vite-extra-56u3yb/src/main.js
Cannot find module '/home/vitejs/stackblitz-reproductions/bluwy-create-vite-extra-56u3yb/node_modules/.pnpm/astro-integration-lottie@0.2.0/node_modules/astro-integration-lottie/dist/types' imported from /home/vitejs/stackblitz-reproductions/bluwy-create-vite-extra-56u3yb/node_modules/.pnpm/astro-integration-lottie@0.2.0/node_modules/astro-integration-lottie/dist/index.js
    at async ESMLoader.import (https://vite-554f.w-credentialless.staticblitz.com/blitz.9bdac2de468a4854f7bacb85c771c93d5676829b.js:6:1209138)
    at async i.loadESM (https://vite-554f.w-credentialless.staticblitz.com/blitz.9bdac2de468a4854f7bacb85c771c93d5676829b.js:6:246622)
    at async handleMainPromise (https://vite-554f.w-credentialless.staticblitz.com/blitz.9bdac2de468a4854f7bacb85c771c93d5676829b.js:6:989144) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```

**Without**
```
❯ node test.js
Error: Cannot find module '/home/vitejs/stackblitz-reproductions/bluwy-create-vite-extra-56u3yb/node_modules/.pnpm/astro-integration-lottie@0.2.0/node_modules/astro-integration-lottie/dist/types' imported from /home/vitejs/stackblitz-reproductions/bluwy-create-vite-extra-56u3yb/node_modules/.pnpm/astro-integration-lottie@0.2.0/node_modules/astro-integration-lottie/dist/index.js
    at async ESMLoader.import (https://vite-554f.w-credentialless.staticblitz.com/blitz.9bdac2de468a4854f7bacb85c771c93d5676829b.js:6:1209138)
    at async i.loadESM (https://vite-554f.w-credentialless.staticblitz.com/blitz.9bdac2de468a4854f7bacb85c771c93d5676829b.js:6:246622)
    at async handleMainPromise (https://vite-554f.w-credentialless.staticblitz.com/blitz.9bdac2de468a4854f7bacb85c771c93d5676829b.js:6:989144) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```

Reply:
Good point, let's just remove the `catch` 👍🏼

Reply:
I think it's better to show the importer info. This is helpful to quickly locate error positions in a big project.

Reply:
I took a look at this locally too yesterday. I think we need the location info, but not in the error message. We can use a WeakMap to save the location message to the error, then re-use the value [here](https://github.com/vitejs/vite/blob/e57cacfb412c7cac9f5e793357df445990d7535f/packages/vite/src/node/ssr/ssrModuleLoader.ts#L205-L217), by calling `logger.error` with the location info before we throw.

I also think the location info message could be improved a bit. As sapphi showed, it's 

```
Failed to import file:///home/vitejs/stackblitz-reproductions/bluwy-create-vite-extra-56u3yb/node_modules/.pnpm/astro-integration-lottie@0.2.0/node_modules/astro-integration-lottie/dist/index.js from /home/vitejs/stackblitz-reproductions/bluwy-create-vite-extra-56u3yb/src/main.js
```

I think ideally it should be something like:

```
Failed to import "astro-integration-lottie" from "src/main.js".
```

Reply:
Instead of `WeakMap`, I bind the imported module id to `error.importee`, then remove it after `logger.error`

BTW, I deduplicate the output ssr-load error.

<img width="1119" alt="image" src="https://user-images.githubusercontent.com/102238922/224526426-2bbede0f-9d95-4cd1-934a-e78313889f03.png">